http-clients/apache-client/src/main/java/software/amazon/awssdk/http/apache/internal/impl/ApacheHttpRequestFactory.java

```diff
@@ -71,7 +71,13 @@ public HttpRequestBase create(final HttpExecuteRequest request, final ApacheHttp
      */
     private String sanitizeUri(URI uri) {
         String newPath = uri.getPath().replace("//", "/%2F");
-        return uri.toString().replace(uri.getPath(), newPath);
+        String baseUri = uri.toString();
+
+        if (baseUri.endsWith("//")) {
```

Comment:
Hmm, you can have a key with leading slash like `/myKey`, so the double slash might not be at the end of the string.

Reply:
Resolved in the latest commit.